connectivity/check/context.go

```diff
@@ -234,7 +234,7 @@ func (ct *ConnectivityTest) Run(ctx context.Context) error {
 
 	// Execute all tests in the order they were registered by the test suite.
 	for _, t := range ct.tests {
-		if isDone(ctx) {
+		if err := ctx.Err(); err != nil {
 			return context.Canceled
```

Comment:
I'm now noticing, this error may swallow up the reason why the context is canceled. 

One could do this:
```
if err := ctx.Err(); err != nil {
    return ctx.Err()
}
```

If you want to keep the nice formating you get by using

```
var Canceled = errors.New("context canceled")
```

We could also error wrap 

```
type ErrCanceled struct {
  e error
}
func (e ErrCanceled) Error() string {
  return "context canceled: " + e.e.Error()
}
// implement unwrap methods if desired...
```

Reply:
I think that the following approach is totally fine:
```go
if err := ctx.Err(); err != nil {
    return ctx.Err()
}
```
If we wanted to error wrap, I'd rather use this simpler approach:
```go
return fmt.Errorf("context done: %w", ctx.Err())
```
Although I'm not sure it would add any value.

Reply:
Agreed. Will update.